src/main/java/org/broadinstitute/hellbender/utils/Utils.java

```diff
@@ -1041,128 +1048,334 @@ public static boolean xor(final boolean x, final boolean y) {
      * @param reference the reference sequence
      * @param query the query sequence
      */
-    public static int lastIndexOf(final byte[] reference, final byte[] query) {
```

Comment:
I would leave lastIndexOf intact and make lastIndexOfAtMostTwoMismatches() its own method that is seperate in the same class.

Reply:
Or at the very least rename it.